internal/api/handler.go

```diff
@@ -0,0 +1,82 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/database"
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+type Handler interface {
+	GetAll(c echo.Context) error
+	ListStories(c echo.Context) error
+	ListJobs(c echo.Context) error
+}
+
+type apiHandler struct {
+	logger   *zap.Logger
+	config   *model.DatabaseConfig
+	dbClient database.Client
+}
+
+type HandlerOptions func(handler *apiHandler)
+
+func NewHandler(ctx context.Context, logger *zap.Logger, config *model.DatabaseConfig, opts ...HandlerOptions) (*apiHandler, error) {
+	handler := &apiHandler{
+		logger: logger,
+		config: config,
+	}
+
+	for _, opt := range opts {
+		opt(handler)
+	}
+
+	if handler.dbClient == nil {
```

Comment:
Would it make sense to just make someone specify this when created the handler, rather than falling back?

Reply:
So ideally I would only create the db client here instead of having the option to pass it in, but since go doesn't support injecting mocks into a constructor, in order for me to test with a mock db client, I inject one with HandlerOptions, but the when just running the api, it will use the default

Reply:
I think I'd prefer one or the other to be honest, either always pass it in or call it new here, rather than a mix for consistency

Reply:
ideally I'd like to just instantiate it as part of the new method but then how would I inject mocks to enable testing?

I don't have to have my main method responsible to instantiating every single client and having to pass them all through. I only want to instantiate a client at the time of needing it

Reply:
In another feature branch, I've ended up moving to one consistent way to do this.

Are you happy to approve on the basis this is going to change in the next iteration?